agent/grpc-external/services/dataplane/get_envoy_bootstrap_params.go

```diff
@@ -73,7 +74,24 @@ func (s *Server) GetEnvoyBootstrapParams(ctx context.Context, req *pbdataplane.G
 		NodeId:      string(svc.ID),
 	}
 
-	bootstrapConfig, err := structpb.NewStruct(svc.ServiceProxy.Config)
+	// This is awkward because it's designed for different requests, but
+	// this fakes the ServiceSpecificRequest so that we can reuse code.
+	_, ns, err := s.MergeNodeServiceWithCentralConfig(
+		nil,
+		store.(*state.Store),
+		&structs.ServiceSpecificRequest{
+			Datacenter:   s.Datacenter,
+			QueryOptions: options,
+		},
+		svc.ToNodeService(),
+		logger,
+	)
+	if err != nil {
+		logger.Error("Error merging with central config", "error", err)
+		return nil, status.Error(codes.Unknown, fmt.Sprintf("Error merging central config: %v", err))
```

Comment:
I could try to discern a better error code here?

Reply:
Seems reasonable to me (since we emit the same code if there is an issue parsing this config below).